examples/src/indexes/vector_stores/singlestore.ts

```diff
@@ -0,0 +1,26 @@
+import { SingleStoreVectorStore } from "langchain/vectorstores/singlestore";
+import { OpenAIEmbeddings } from "langchain/embeddings/openai";
+import { createPool } from "mysql2/promise";
+
+export const run = async () => {
+  const pool = createPool({
+    host: process.env.SINGLESTORE_HOST,
+    port: Number(process.env.SINGLESTORE_PORT),
+    user: process.env.SINGLESTORE_USERNAME,
+    password: process.env.SINGLESTORE_PASSWORD,
+    database: process.env.SINGLESTORE_DATABASE,
+  });
```

Comment:
I think generally I'd have a preference for making `mysql2` an implementation detail within your store - basically passing in `connectionParams` rather than passing in the entire `connectionPool`. Fewer imports for the developer, and allows you to swap packages more seamlessly if something should happen that dependency. 

But up to you! Looks good otherwise and will merge by tomorrow with or without that change.

Reply:
The `createPool` allows a developer to pass many different parameters (not only those four that I used in the example) depending on the type of connection. I'd prefer to keep this flexibility and not copy-paste all parameters as parameters of the vector store.

Reply:
They may export a type like `ConnectionPoolParams` with all options that you can then initialize a pool with. No worries though, will merge shortly.